test/jdk/java/util/Calendar/RollHoursTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8367901
+ * @summary Ensure hour rolling is correct. Particularly, when the HOUR/HOUR_OF_DAY
+ *          amount rolled would cause the calendar to originate on the same hour as before
+ *          the call.
+ * @run junit RollHoursTest
+ */
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.FieldSource;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RollHoursTest {
+
+    // Should trigger multiple full HOUR/HOUR_OF_DAY cycles
+    private static final List<Integer> hours =
+            IntStream.rangeClosed(-55, 55).boxed().toList();
+    // Sample AM/PM Hours
+    private static final int AM_HOUR = 3;
+    private static final int PM_HOUR = 15;
+
+    // Roll the HOUR_OF_DAY field. AM/PM specific calendar does not matter.
+    @ParameterizedTest
+    @FieldSource("hours")
+    void HourOfDayTest(int hour) {
+        var cal = new GregorianCalendar(2005, 8, 12, PM_HOUR, 30, 45);
```

Comment:
IIUC, the regression was caused by the fix to [JDK-8152077](https://bugs.openjdk.org/browse/JDK-8152077), where it jiggles an hour on the transition day. So although I don't think it would fail, but it might be helpful to test on one of those days